lib/runtime_tools/src/dbg.erl

```diff
@@ -2163,38 +2261,43 @@ trac(Node, {_Replay, Tracer}, AtomPid, How, Flags) ->
     end.
 
 -doc false.
-erlang_trace(AtomPid, How, Flags) ->
+erlang_trace(Session, AtomPid, How, Flags) ->
     case to_pidspec(AtomPid) of
 	{badpid,_} ->
 	    {no_proc,AtomPid};
-	P ->
-	    erlang:trace(P, How, Flags)
+	P when is_pid(P) ->
+            trace:process(Session, P, How, Flags);
+        P when is_port(P) ->
+            trace:port(Session, P, How, Flags);
+        P ->
+            trace:process(Session, P, How, Flags) +
+            trace:port(Session, P, How, Flags)
```

Comment:
I don't think this will work as `trace:process` will not accept *ports atoms and `trace:port` will not accept *processes atoms. Maybe they _should_ silently ignore such atoms?

Reply:
Calling `trace:process(S, all, true, [ports])` works, but returns 0, so it seems to already today be ignored?

Reply:
I mean these will currently fail:
```
trace:process(S, ports, ...).
trace:process(S, new_ports, ...).
trace:process(S, existing_ports, ...).
trace:port(S, processes, ...).
trace:port(S, new_processess, ...).
trace:port(S, existing_processes, ...).
```